nixd/include/nixd/Server/Controller.h

```diff
@@ -41,6 +41,12 @@
 
 namespace nixd {
 
+struct OptionalValue {
```

Comment:
how about just type-def this optional value?

```
using OptionalValue = std::optional<llvm::json::Value> Value;
```

Reply:
If I did this, then the new `fromJSON` function would apply to all `std::optional<llvm::json::Value>` types across the program, which I don't think is a good idea as I think `std::optional<T>` should instead be used to handle `null` in the general case.